src/components/JournalEntry/EditJournalEntry.tsx

```diff
@@ -124,15 +126,71 @@ const extensions = [
   StarterKit,
 ];
 
-export const EditJournalEntry = () => {
+export const EditJournalEntry = ({
+  onSave,
+}: {
+  onSave: (boxContent: JSONContent) => void;
+}) => {
+  const editorRef = useRef<Editor | null>(null);
+
+  const [wizard, setWizard] = useState<
+    | {
+        type: "initial";
+      }
+    | {
+        type: "loading";
+      }
+    | {
+        type: "error";
+        message: string;
+      }
+  >({ type: "initial" });
+
+  const sendEntry = () => {
+    setWizard({ type: "loading" });
+    const saveAndClear = () => {
+      if (editorRef.current !== null && editorRef.current.isEmpty === false) {
+        onSave(editorRef.current.getJSON());
+        editorRef.current.commands.clearContent();
+      }
+      setWizard({ type: "initial" });
+    };
+
+    const entryPromise = new Promise<void>((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, 1000);
+    });
+
+    entryPromise.then(() => {
+      saveAndClear();
+    });
+  };
```

Comment:
The promise should be in the parent, because the parent is the one who will call the API. 

The code in this function should be like this. Use this code as helper, to write the parent code.

```js
try {
  setWizard({ type: "loading" });
  await onSave(editorRef.current.getJSON())
  // clear content
  // set state back to initial
} catch (err) {
  // set error state
}
```